File: common/src/main/java/dev/obscuria/fragmentum/network/FragmentumNetworking.java
package dev.obscuria.fragmentum.network;

import dev.obscuria.fragmentum.Fragmentum;
import net.minecraft.core.BlockPos;
import net.minecraft.server.MinecraftServer;
import net.minecraft.server.level.ServerLevel;
import net.minecraft.server.level.ServerPlayer;
import net.minecraft.world.entity.Entity;

@SuppressWarnings("unused")
public interface FragmentumNetworking
{
    static PayloadRegistrar registrar(String modId)
    {
        return Fragmentum.SERVICES.network().payloadRegistrar(modId);
    }

    static <T> void reply(T payload)
    {
        Fragmentum.SERVICES.network().reply(payload);
    }

    static <T> void sendTo(ServerPlayer player, T payload)
    {
        Fragmentum.SERVICES.network().sendTo(player, payload);
    }

    static <T> void sendToTracking(ServerLevel level, BlockPos pos, T payload)
    {
        Fragmentum.SERVICES.network().sendToTracking(level, pos, payload);
    }

    static <T> void sendToTracking(Entity entity, T payload)
    {
        Fragmentum.SERVICES.network().sendToTracking(entity, payload);
    }

    static <T> void sendToAll(MinecraftServer server, T payload)
    {
        Fragmentum.SERVICES.network().sendToAll(server, payload);
    }

    static <T> void sendToServer(T payload)
    {
        Fragmentum.SERVICES.network().sendToServer(payload);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/api/network/FragmentumNetworking.kt
package dev.obscuria.fragmentum.api.network

import dev.obscuria.fragmentum.api.Fragmentum
import net.minecraft.core.BlockPos
import net.minecraft.server.MinecraftServer
import net.minecraft.server.level.ServerLevel
import net.minecraft.server.level.ServerPlayer
import net.minecraft.world.entity.Entity

object FragmentumNetworking
{
    fun registrar(modId: String): PayloadRegistrar =
        Fragmentum.SERVICES.network().payloadRegistrar(modId)

    fun <T : Any> reply(payload: T) =
        Fragmentum.SERVICES.network().reply(payload)

    fun <T : Any> sendTo(player: ServerPlayer, payload: T) =
        Fragmentum.SERVICES.network().sendTo(player, payload)

    fun <T : Any> sendToTracking(world: ServerLevel, pos: BlockPos, payload: T) =
        Fragmentum.SERVICES.network().sendToTracking(world, pos, payload)

    fun <T : Any> sendToTracking(entity: Entity, payload: T) =
        Fragmentum.SERVICES.network().sendToTracking(entity, payload)

    fun <T : Any> sendToAll(server: MinecraftServer, payload: T) =
        Fragmentum.SERVICES.network().sendToAll(server, payload)

    fun <T : Any> sendToServer(payload: T) =
        Fragmentum.SERVICES.network().sendToServer(payload)
}